docs/src/account/id.md

```diff
@@ -0,0 +1,63 @@
+# Account ID
+
+> [!Note]
+> An immutable and unique identifier for the `Account`.
+
+The `Account` ID is a 120-bit long number. This identifier is designed to contain the metadata of an account. The metadata includes the [account type](#account-type), [account storage mode](#account-storage-mode) and the version of the `Account`. This metadata is included in the ID to ensure it can be read without needing the full account state.
+
+The ID is generated by hashing a randomly generated seed together with commitments to the initial code and storage of the `Account` until the resulting ID has the desired account type and storage mode. This process requires a small amount of Proof-of-Work (9 bits) that can be done even by low-powered devices. The resulting 256-bit hash is shortened to 120 bits.
+
+Account type and storage mode are chosen at account creation time and cannot be changed later.
+
+### Account type
+
+There are two main categories of accounts in Miden: **basic accounts** and **faucets**.
+
```

Comment:
This doesn't align very well with the `AccountType` enum we have, but I think that's not such an issue. 

More broadly though, this might benefit from a discussion on the naming: "Basic Account" doesn't reflect the programmable aspect of it (in fact, a "Basic Account" code can get quite complex). While the distinction is very different from EOAs and Contract accounts in Ethereum, I like how they are descriptive. This is probably not for this PR but I think worth bringing up :)

Reply:
Yeah, I don't love the naming either. "Faucet" is actually fine (I think), but not having a good name for a non-faucet account is kind of annoying. Maybe it could be just "Faucets" and "Accounts", but then not sure what the supercategory of these would be.